receiver/awscloudwatchmetricsreceiver/config.go

```diff
@@ -0,0 +1,179 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awscloudwatchmetricsreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscloudwatchmetricsreceiver"
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"go.uber.org/multierr"
+)
+
+var (
+	defaultPollInterval = time.Minute
+)
+
+// Config is the overall config structure for the awscloudwatchmetricsreceiver
+type Config struct {
+	Region       string         `mapstructure:"region"`
+	Profile      string         `mapstructure:"profile"`
+	IMDSEndpoint string         `mapstructure:"imds_endpoint"`
+	PollInterval time.Duration  `mapstructure:"poll_interval"`
+	Metrics      *MetricsConfig `mapstructure:"metrics"`
+}
+
+// MetricsConfig is the configuration for the metrics part of the receiver
+// added this so we could expand to other inputs such as autodiscover
+type MetricsConfig struct {
+	Names []*NamedConfig `mapstructure:"named"`
+}
+
+// NamesConfig is the configuration for the metric namespace and metric names
+// https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html
+type NamedConfig struct {
+	Namespace      string                   `mapstructure:"namespace"`
+	MetricName     string                   `mapstructure:"metric_name"`
+	Period         time.Duration            `mapstructure:"period"`
+	AwsAggregation string                   `mapstructure:"aws_aggregation"`
+	Dimensions     []MetricDimensionsConfig `mapstructure:"dimensions"`
+}
+
+// MetricDimensionConfig is the configuration for the metric dimensions
+type MetricDimensionsConfig struct {
+	Name  string `mapstructure:"Name"`
+	Value string `mapstructure:"Value"`
+}
+
+var (
+	errNoMetricsConfigured = errors.New("no metrics configured")
+	errNoRegion            = errors.New("no region was specified")
+	errInvalidPollInterval = errors.New("poll interval is incorrect, it must be a duration greater than one second")
+
+	// https://docs.aws.amazon.com/cli/latest/reference/cloudwatch/get-metric-data.html
+	// GetMetricData supports up to 500 metrics per API call
+	errTooManyMetrics = errors.New("too many metrics defined")
```

Comment:
Maybe we shall add the information about the limit to the error message. You've done that for `errTooManyDimensions`, so let's add it here too.

Reply:
Fixed in [8ab9607](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/19429/commits/8ab9607cd8a44fe14c0e9f55646914d57d3ee8fb)